### PR TITLE
Registry driver version info

### PR DESCRIPTION
**Pull request by igorbrsn**
**The problem**
We need alternative solution to retrieve driver info in case AGS library would fail to retrieve it   

**The solution**
Read registry from `HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\DirectX\*` and write it to file 

**Result**
We can use it as an alternative way to retrieve driver.

Example of registry info
NVIDIA driver info

### CODE REVIEW

File: framework/util/driver_info.cpp [hunk 1]
```diff
@@ -172,6 +281,48 @@ bool GetDriverInfo(std::string& driver_info, format::ApiFamilyId api_family)
...
+std::string ConvertDWORDtoVersionNumber(uint64_t dword)
+{
+    uint32_t    version[4]  = {};
+    std::string str_version = "";
+    version[0]              = (unsigned int)((dword & 0xFFFF000000000000) >> 16 * 3);
+    version[1]              = (unsigned int)((dword & 0x0000FFFF00000000) >> 16 * 2);
+    version[2]              = (unsigned int)((dword & 0x00000000FFFF0000) >> 16 * 1);
+    version[3]              = (unsigned int)((dword & 0x000000000000FFFF));
+    str_version = std::to_string(version[0]) + "." + std::to_string(version[1]) + "." + std::to_string(version[2]) +
+                  "." + std::to_string(version[3]) + ".";
```
Comment (davidd-lunarg) [inline framework/util/driver_info.cpp:315]: Final period in version number could be removed: `Driver Version: 31.0.12029.4013.` -> `Driver Version: 31.0.12029.4013`

**Review by davidd-lunarg — CHANGES_REQUESTED** [review]
If possible I think it would be good to capture any available driver info and include it in the output of `gfxrecon-info`. For example, when `AMD_IsDriverActive` is true include both AMD and system version info:
```
Driver info:
        AMD* Radeon software version: 22.10.3
        AMD* driver version: 22.20.29.04-221025a-385333C-AMD-Software-Adrenalin-Edition
        AMD* UMD version (amdxc64.dll): 8.18.10.0462 
        System driver version: 31.0.12029.4013
        System UMD version: 31.0.12029.4013
```
(\* Can use `AMD` or whatever desired prefix for these values.)

and when `AMD_IsDriverActive` is false:
```
Driver info:
        System driver version: 31.0.12029.4013
        System UMD version: 31.0.12029.4013
```

This gives any tools that might parse the output of `gfxrecon-info` a consistent field (e.g., `System driver version`) to look for. It also adds a new version # not currently included when `AMD_IsDriverActive` is true.

File: framework/util/driver_info.cpp [hunk 2]
```diff
@@ -172,6 +281,48 @@ bool GetDriverInfo(std::string& driver_info, format::ApiFamilyId api_family)
...
+            driver_info_found = true;
+        }
+    }
+
+    return driver_info_found;
+}
+
+#endif
+
+std::string ConvertDWORDtoVersionNumber(uint64_t dword)
```
Comment (andrew-lunarg) [inline framework/util/driver_info.cpp:306]: I think a DWORD is still 32 bits on 64 bit Windows. Does this function and the argument to it need tweaked names?
https://learn.microsoft.com/en-gb/windows/win32/winprog/windows-data-types
https://learn.microsoft.com/en-us/windows/win32/winprog64/the-new-data-types
https://godbolt.org/z/4sbjG85nb

Comment (igorbrsn) [reply framework/util/driver_info.cpp]: fixed

**Review by davidd-lunarg — APPROVED** [review]

**Pull request merged**